tests/unit/projection-test.js

```diff
@@ -0,0 +1,1464 @@
+      });
+
+      let {
+        baseRecordWatcher,
+      } = this.watchers;
+
+      let baseCounters = baseRecordWatcher.counters;
+
+      assert.watchedPropertyCount(baseCounters.description, 1, 'Afterwards we have dirtied baseRecord.description');
+      assert.equal(get(baseRecord, 'description'), NEW_DESCRIPTION, 'base-record has the correct description');
+    });
+
+    test('Updating the base-record updates projections', function(assert) {
+      let store = this.store();
+      let {
+        baseRecord,
+      } = this.records;
+
+      run(() => {
+        store.push({
+          data: {
+            id: BOOK_ID,
+            type: BOOK_CLASS_PATH,
+            attributes: {
+              title: NEW_TITLE,
+              'chapter-1': NEW_CHAPTER_TEXT,
+              description: NEW_DESCRIPTION,
+            },
+          },
+        });
+      });
+
+      let {
+        baseRecordWatcher,
+      } = this.watchers;
+
+      let baseCounters = baseRecordWatcher.counters;
+
+      assert.watchedPropertyCount(baseCounters.description, 1, 'Afterwards we have dirtied baseRecord.description');
+      assert.equal(get(baseRecord, 'description'), NEW_DESCRIPTION, 'base-record has the correct description');
+    });
+
+    test('Setting a projection updates the base-record and other projections', function(assert) {
+      let preview = this.records.projectedPreview;
+      let baseRecord = this.records.baseRecord;
+
+      run(() => {
+        set(preview, 'chapter-1', NEW_CHAPTER_TEXT);
+        set(preview, 'title', NEW_TITLE);
+      });
+
+      assert.throws(() => {
+        run(() => { set(preview, 'description', NEW_DESCRIPTION); });
+      }, /whitelist/gi, 'Setting a non-whitelisted property throws an error');
+      assert.watchedPropertyCount(this.watchers.baseRecordWatcher.counters.description, 0, 'Afterwards we have not dirtied baseRecord.description');
+      assert.equal(get(baseRecord, 'description'), BOOK_DESCRIPTION, 'base-record has the correct description');
+    });
+
+    test('Updating a projection updates the base-record and other projections', function(assert) {
+      let baseRecord = this.records.baseRecord;
+      let store = this.store();
+
+      run(() => {
+        store.push({
+          data: {
+            id: BOOK_ID,
+            type: BOOK_EXCERPT_PROJECTION_CLASS_PATH,
+            attributes: {}
+          },
+          included: [{
+            id: BOOK_ID,
+            type: BOOK_CLASS_PATH,
+            meta: {
+              partial: true
+            },
+            attributes: {
+              title: NEW_TITLE,
+              'chapter-1': NEW_CHAPTER_TEXT,
+              /*
+                The below update is invalid because in the real world the schema was are given is also used
+                to create the payload the API gives us, so we could not have properties from the API that don't
+                exist in the whitelist.
+               */
```

Comment:
is there any reason for m3 to have this restriction?

Reply:
There is no restriction for M3, I think it is a restriction of the test as it expects description to not have been changed, but we dirtied irrespective of the fact the value is the same.

Reply:
ok let's remove the comment then.  As it stands, it looks like it's saying including `description` would be invalid generally, rather than an artifact of this particular set of tests.

Reply:
I'll refactor it, basically this is a comment that made more sense when the payloads weren't aligned closer to the concept of model-data: at that time we would push payloads from the perspective of individual projections, but pushing in data for a projection that would never have that data didn't make sense.